byterun/caml/mlvalues.h

```diff
@@ -25,6 +25,12 @@ typedef intnat value;
 typedef int32 opcode_t;
 typedef opcode_t * code_t;
 
+/* atomic_uintnat is declared as a struct type to prevent accidental
+   accesses that don't use the primitives of platform.h */
+typedef struct atomic_uintnat {
+  volatile uintnat val;
+} atomic_uintnat;
```

Comment:
I know it's just a move in this commit, but you may want to consider padding this with an architecture-specific cache line size minus sizeof(uintnat), and to [align to 64 bytes](https://stackoverflow.com/questions/7281699/aligning-to-cache-line-and-knowing-the-cache-line-size). Don't want atomics sharing cache lines, especially with other atomics.